modules/onewire/onewire.c

```diff
@@ -65,6 +66,18 @@ OneWireStatus OneWire_Init(OneWire_t* OneWireStruct, uint32_t PalLine) {
     return ONEWIRE_SUCCESS;
 }
 
+temp_sensor_status_t OneWire_System_Init(temp_config_t* temp_config){

```

Comment:
Please correct me if I'm wrong...

onewire.c is going to define operations using onewire protocol.  onewire.c won't (or shouldn't) care about what kind of device it's connected to, right?  If so, then having this function coupled to a sensor seems out of place.

Reply:
I'd agree on this! Ultimately `onewire` should have no notion of `ds18b20` or `temp_sensor`, nor should `ds18b20` have one of `temp_sensor, nor should `temp_sensor` of `onewire`.Would think about usage being uni-directional: `temp_sensor` uses `ds18b20` uses `onewire`. Things get overly intertwined if `temp_sensor` <--> `ds18b20` <--> `onewire`.

Reply:
I agree as well. This has since been fixed